### PR TITLE
cmd/faucet, cmd/puppeth: support multi-tiered faucet

### PR DESCRIPTION
**Pull request by karalabe**
Different people have different work styles. Some like to request faucet funds more frequently, some want a big batch that they can then spend without worrying about more. This PR adds a tiered feature to the faucet that supports retrieving bigger amounts of funds in exchange for longer request timeout.

There is a slight preference towards more frequent requests via a funding penalty on large amounts.

### CODE REVIEW

File: cmd/faucet/faucet.go [hunk 1]
```diff
@@ -89,22 +92,46 @@ func main() {
...
+			amounts[i] = strings.TrimSuffix(amounts[i], "s")
+		}
+		// Calcualte the period for th enext tier and format it
+		period := *minutesFlag * int(math.Pow(3, float64(i)))
+		periods[i] = fmt.Sprintf("%d mins", period)
+		if period%60 == 0 {
+			period /= 60
+			periods[i] = fmt.Sprintf("%d hours", period)
+		}
+		if period%24 == 0 {
```
Comment (holiman) [inline cmd/faucet/faucet.go]: I don't follow.. So `period` is expressed in minutes. So what if it's set to 24 minutes - divisable by `24` but not by `60` ? Don't you need to always divide by `60`, not just when it's divisable by `60` ?

Comment (karalabe) [inline cmd/faucet/faucet.go]: Whoops. Good catch!

Comment (karalabe) [inline cmd/faucet/faucet.go]: I added this last minute since until now we didn't care about days. Fil fix in a moment.

Comment (karalabe) [inline cmd/faucet/faucet.go]: Pushed a commit to fix it.

**Pull request merged**